Fixed. #2の課題が完了しました。

TsunakiM opened the pull request.

TsunakiM committed:
Fixed. #2の課題が完了しました。

TsunakiM committed:
Fix. #2の課題を修正しました。

ファイル名を日本語からアルファベットに変更。

TsunakiM committed:
Fix. #2の課題を修正しました。

ファイル名を日本語からアルファベットに変更。

TsunakiM committed:
Merge branch 'TsunakiMasuda/1-1-2' of https://github.com/TsunakiMasuda/objc-basic into TsunakiMasuda/1-1-2

TsunakiM committed:
Fix. #2のコンフリクトの解消。

TsunakiM committed:
Fix. コンフリクトの解消。

TsunakiMasuda commented:
ファイル名を訂正しました。
（日本語→アルファベット）

taguchi-k commented on traning1-1-2/traning1-1-2/main.m:
```diff
@@ -0,0 +1,40 @@
+    @autoreleasepool {
+        // 1-1-2 NSArray型、NSDictonary型の変数を定義し、コンソールにログを出力する。(NSLog)
+        
+        NSArray *arrayA = [NSArray arrayWithObjects: @"weather", @"晴れ", @"曇り", @"雨", @"雪", nil];
```
レガシーな記法になっているので、モダン記法に修正してください。

taguchi-k replied:
`arrayA`だと何を表す変数なのかわからないので、適切な変数名に修正してください。

taguchi-k commented on traning1-1-2/traning1-1-2/main.m:
```diff
@@ -0,0 +1,40 @@
+        // 1-1-2 NSArray型、NSDictonary型の変数を定義し、コンソールにログを出力する。(NSLog)
+        
+        NSArray *arrayA = [NSArray arrayWithObjects: @"weather", @"晴れ", @"曇り", @"雨", @"雪", nil];
+        NSDictionary *dicB = [NSDictionary dictionaryWithObjectsAndKeys:
```
レガシーな記法になっているので、モダン記法に修正してください。

taguchi-k replied:
`dicB `だと何を表す変数なのかわからないので、適切な変数名に修正してください。

taguchi-k commented on traning1-1-2/traning1-1-2/main.m:
```diff
@@ -0,0 +1,40 @@
+                              @"前橋市", @"群馬県",
+                              nil];
+        // int型だとXcodeから警告が出るため、厳密にchar型を使用
+        char dicLength = [dicB count];
```
`count`の型は`NSUInteger`です。
プロパティやメソッドに関する情報はAppleのリファレンスを参照しましょう。
https://developer.apple.com/documentation/foundation/nsdictionary/1409628-count?language=objc

taguchi-k replied:
`count`はプロパティなのでドットアクセスするようにしてください。

taguchi-k replied:
`count`は辞書内の要素数を表すプロパティなので、`dicLength`は適切ではありません。

taguchi-k commented on traning1-1-2/traning1-1-2/main.m:
```diff
@@ -0,0 +1,40 @@
+        // Dictionaryの長さ（個数）の表示
+        NSLog(@"NSDictionary.length: %d", dicLength);
+        // キー（都道府県）から、値（県庁所在地）を出す
+        NSLog(@"NSDictionary: 東京都の県庁所在地は %@", [dicB objectForKey:@"東京都"]);
```
要素の取得がレガシーな記法になっているので、モダン記法に修正してください。

taguchi-k commented on traning1-1-2/traning1-1-2/main.m:
```diff
@@ -0,0 +1,40 @@
+        NSLog(@"NSDictionary.length: %d", dicLength);
+        // キー（都道府県）から、値（県庁所在地）を出す
+        NSLog(@"NSDictionary: 東京都の県庁所在地は %@", [dicB objectForKey:@"東京都"]);
+        NSLog(@"NSDictionary: 神奈川県の県庁所在地は %@", [dicB objectForKey:@"神奈川県"]);
```
要素の取得がレガシーな記法になっているので、モダン記法に修正してください。

taguchi-k reviewed (changes requested):
@TsunakiMasuda 
レビューしました。確認・対応お願いします。

TsunakiM committed:
Fix. #2の課題を修正しました。

レガシー記法をモダン記法に書き換え。
変数名を適正化（arrayA => weatherList, dicB => capitalDic）。
countの型を、charからNSUintegerに適正化。

TsunakiMasuda commented:
@taguchi-k レビューありがとうございます。ご指摘頂いた部分を修正しましたので、再度レビューをお願いします。

taguchi-k commented on traning1-1-2/traning1-1-2/main.m:
```diff
@@ -0,0 +1,41 @@
+        // 1-1-2 NSArray型、NSDictonary型の変数を定義し、コンソールにログを出力する。(NSLog)
+        
+        NSArray *weatherList = @[@"weather", @"晴れ", @"曇り", @"雨", @"雪"];
+        NSDictionary *capitalDic = @{
```
`Dic`と省略せずに`Dictionary`としましょう。
Appleがそのような書き方を推奨しているためです。

例外もあるので詳しくは下記リンク先の「使って構わない略語や頭字語」を参照してください。
https://developer.apple.com/jp/documentation/CodingGuidelines.pdf

taguchi-k commented on traning1-1-2/traning1-1-2/main.m:
```diff
@@ -0,0 +1,41 @@
+                                     @"群馬県": @"前橋市",
+                                     };
+
+        NSUInteger dicCount = capitalDic.count;
```
`dic`と省略せずに`dictionary`としましょう。
理由は先述のとおりです。

taguchi-k reviewed (changes requested):
@TsunakiMasuda 
だいぶいい感じになりました！
再度コメントしたので確認・対応お願いします。

TsunakiM committed:
Fix. #3の課題を修正（変数名の変更）

capitalDicを、capitalDictionaryに変更。

TsunakiMasuda commented:
@taguchi-k 課題1-1-2、再レビューありがとうございます。
ご指摘頂いた部分を修正しましたので、再度レビューをお願いします。

また、コーディングガイドラインのご提示、ありがとうございます。

taguchi-k commented:
@TsunakiMasuda 
レビューOKです。お疲れ様でした！
プルリクマージして、issuesをクローズしてください。
